### PR TITLE
Upgrade dep (drop Python 2 support) & refactor all tests to use python instead of yaml <=> more flexible

### PR DESCRIPTION
**Pull request by ryarnyah**
## Proposed Changes

  - Upgrade test dependencies
  - Remove Yaml tests to use pytest

### CODE REVIEW

**Comment by ryarnyah** [comment]
This is a proposition to drop Python 2 support on tests and to have only best effort support for it. I will check if we can continue with the test refacto without Python 3 but it may be time to drop Python 2 support...

**Comment by StephenSorriaux** [comment]
> This is a proposition to drop Python 2 support on tests and to have only best effort support for it. I will check if we can continue with the test refacto without Python 3 but it may be time to drop Python 2 support...

I am perfectly fine with dropping Python 2, it is already EOL and now can be the time to let it go!

File: README.md [hunk 1]
```diff
@@ -374,13 +374,14 @@ cacert_content: |
 ```
 ## Python compatibility
 This library is tested with the following versions of Python:
-* Python 2.7
+* Python 2.7 (last version 0.10.0)
```
Comment (StephenSorriaux) [inline README.md:377]: I just tagged a `0.11.0` version, is it ok to update it here?

File: README.md [hunk 2]
```diff
@@ -374,13 +374,14 @@ cacert_content: |
...
 * Python 3.6
+* Python 3.7
```
Comment (StephenSorriaux) [inline README.md:379]: Also Python 3.8 I guess (and even Python 3.9 I hope)?

**Review by StephenSorriaux — APPROVED** [review]
Thanks a lot for this!

File: .github/workflows/python-package.yml [hunk 3]
```diff
@@ -13,7 +13,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [2.7, 3.6, 3.7]
+        python-version: [3.6, 3.7, 3.8]
```
[the remark this answers is not in the document]
Comment (ryarnyah) [reply .github/workflows/python-package.yml]: Done

File: README.md (hunk 1 above)
Comment (ryarnyah) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (ryarnyah) [reply README.md:379]: Done

**Pull request merged**